tests/test_agent_output_schemas.py

```diff
@@ -0,0 +1,630 @@
+import pytest
+from inline_snapshot import snapshot
+from pydantic import BaseModel
+
+from pydantic_ai import (
+    Agent,
+    BinaryImage,
+    DeferredToolRequests,
+    NativeOutput,
+    PromptedOutput,
+    StructuredDict,
+    ToolOutput,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+class Bar(BaseModel):
+    answer: str
+
+
+class Foo(BaseModel):
+    a: list[Bar]
+    b: int
+
+
+async def test_text_output_json_schema():
+    agent = Agent('test')
+    assert agent.output_json_schema() == snapshot({'type': 'string'})
+
+
+async def test_auto_output_json_schema():
+    agent = Agent('test', output_type=bool)
+    assert agent.output_json_schema() == snapshot(
+        {'type': 'object', 'properties': {'response': {'type': 'boolean'}}, 'required': ['response']}
```

Comment:
Looks like we need some special handling for the `response` key added here:

https://github.com/pydantic/pydantic-ai/blob/c27e5e4e32f16d2e28c3b25a10528f64335e7cfa/pydantic_ai_slim/pydantic_ai/_output.py#L564-L580

Because it's unwrapped when validating the output:

https://github.com/pydantic/pydantic-ai/blob/c27e5e4e32f16d2e28c3b25a10528f64335e7cfa/pydantic_ai_slim/pydantic_ai/_output.py#L670-L671

Meaning that the real output schema is (and result of this method should be) `{'type': 'boolean'}`